web/src/banners.ts

```diff
@@ -42,6 +42,32 @@ export function close($banner: JQuery): void {
     $(window).trigger("resize");
 }
 
+export function set_banner_progress($banner: JQuery, value: number, max: number): void {
+    if (max <= 0) {
+        throw new Error("Maximum value must be greater than 0");
+    }
+
+    if (value < 0 || value > max) {
+        throw new Error(`Value (${value}) must be between 0 and maximum (${max})`);
+    }
+
+    let $progress_bar = $banner.find(".banner-progress-bar");
+    if ($progress_bar.length === 0) {
+        $progress_bar = $("<span>").addClass("banner-progress-bar");
+        $banner.prepend($progress_bar);
```

Comment:
Prepending inside the banner root is a little awkward. Is there a reason the original insertion of the progress bar element shouldn't instead be an optional template feature?

Reply:
Yeah we could do the progress bar as an optional template feature, I was trying to make the progress feature dynamic this way but I think nothing great is achieved doing that since it is just an empty span.